### PR TITLE
Add more perl signatures in lib:OpenQA

### PR DESCRIPTION
**Pull request by FruitFly638**
Add perl signatures to lib/openQA/ Command.pm, Events.pm and Files.pm

https://progress.opensuse.org/issues/105127

### CODE REVIEW

File: lib/OpenQA/File.pm [hunk 1]
```diff
@@ -79,8 +76,7 @@ sub get_piece {
     return $self->_chunk($index, $n_chunks, $chunk_size, $residual, $total_cksum);
 }
 
-sub split {
-    my ($self, $chunk_size) = @_;
+sub split ($self, $chunk_size = undef) {
```
Comment (baierjan) [inline lib/OpenQA/File.pm:79]: Shouldn't this one be `$chunk_size = 10000000`?

Comment (perlpunk) [reply lib/OpenQA/File.pm:79]: Signatures have one problem that can be easily overlooked: the default value will only be assigned if there was no value passed, e.g. if the argument list was empty her e(except `$self`).
So if someone passes undef by passing a variable, e.g.
`$file->split($some_var_that_might_be_undef)`
then the default assignment won't happen.

So passing `undef` here is the safest way.

**Review by perlpunk — APPROVED** [review]

File: lib/OpenQA/Command.pm [hunk 2]
```diff
@@ -101,8 +92,7 @@ sub run {
     return $self->command(@args);
 }
 
-sub url_for {
-    my ($self, $path) = @_;
+sub url_for ($self, $path) {
```
Comment (perlpunk) [inline lib/OpenQA/Command.pm:95]: hm, `url_for` https://docs.mojolicious.org/Mojolicious/Controller#url_for might be called with more arguments.
Let's see if tests are failing.

Comment (perlpunk) [reply lib/OpenQA/Command.pm:95]: I just tested it locally and replaced
`my $url = $self->url_for($path);`
with
`my $url = $self->url_for($path, { x => 23 });`
in `lib/OpenQA/CLI/api.pm` and calling `perl script/openqa-cli api jobs/1 --o3 ` and it died.
So better add a `@` here.

**Review by perlpunk — CHANGES_REQUESTED** [review]
See comment

File: lib/OpenQA/Command.pm (hunk 2 above)
Comment (kraih) [reply lib/OpenQA/Command.pm:95]: There's no relation to the Mojolicious `url_for` method though.

Comment (perlpunk) [reply lib/OpenQA/Command.pm:95]: oh I see. then ignore my comment

**Review by perlpunk — APPROVED** [review]

File: lib/OpenQA/Command.pm (hunk 2 above)
Comment (kraih) [reply lib/OpenQA/Command.pm:95]: Bit of a confusing naming choice here, no idea why i did that. 🤷

File: lib/OpenQA/File.pm (hunk 1 above)
Comment (FruitFly638) [reply lib/OpenQA/File.pm:79]: True, then I could just delete the other line

File: lib/OpenQA/Command.pm (hunk 2 above)
Comment (FruitFly638) [reply lib/OpenQA/Command.pm:95]: Yeah, my local tests didnt fail but I wanted to see if tests fail here

**Review by okurz — CHANGES_REQUESTED** [review]
It was ok for the former PRs but now please really rename the git subject line to be unique. We use the git commit message subject lines for the package changelogs and the duplicate entries "Add more perl signatures" is already not nice, see https://build.opensuse.org/package/view_file/devel:openQA/openQA/_service:obs_scm:openQA.changes?expand=1

**Comment by FruitFly638** [comment]
> It was ok for the former PRs but now please really rename the git subject line to be unique. We use the git commit message subject lines for the package changelogs and the duplicate entries "Add more perl signatures" is already not nice, see https://build.opensuse.org/package/view_file/devel:openQA/openQA/_service:obs_scm:openQA.changes?expand=1

I made some changes, is this better? including the directory makes it more unique

**Review by kalikiana — APPROVED** [review]

**Review by okurz — COMMENTED** [review]
Well, better but nearly all files are in this dir. Better use OpenQA::Commands and only include this one file in one commit and make separate ones for the other file

**Review by okurz — APPROVED** [review]

**Pull request merged**